R/mash_wrapper.R

```diff
@@ -470,16 +470,59 @@ load_multitrait_R_sumstat <- function(
   }
 
   merge_matrices <- function(matrix_list, value_column, id_column = "variants",
-                             remove_any_missing = FALSE) {
-    # Convert matrices to data frames
-    df_list <- lapply(seq_along(matrix_list), function(i) {
+                             remove_any_missing = FALSE, ld_ref = ld_ref, allele_flip = TRUE) {
+    df_list <- lapply(seq_along(matrix_list), function(i){
+      # Step 1: Convert matrix to data frame and extract relevant columns
       df <- as.data.frame(matrix_list[[i]])
       df2 <- df[, c(id_column, value_column)]
-      # Rename columns to avoid duplication
-      colnames(df2) <- c(id_column, paste0(value_column, "_", i))
-      return(df2)
-    })
-
+      if (allele_flip) {
+       # Step 2: Split 'variants' to extract chromosomal info
```

Comment:
can you define this as a function called `parse_variant_id()` similar to our `parse_region` and put it right under: https://github.com/cumc/pecotmr/blob/main/R/misc.R#L218 ?